developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -2092,6 +2092,30 @@ Similarly for mutations:
 
 where `::Mutations::Ducks::Delete` is your delete mutation class inheriting from `::Mutations::DeleteMutation`.
 
+[[adding-subscribers]]
+=== Adding subscribers
+
+_Requires Foreman 2.0 or higher, set requires_foreman '>= 2.0' in
+engine.rb_
+
+You can consume events from Foreman core by registering subscribers. Subscriber is a class that inherits `::BaseSubscriber`. It is recommended to store subscribers under the `/app/subscribers/my_plugin/` directory. If you have your subscriber defined register it in the plugin. Example of your engine.rb:
+
+[source, ruby]
+....
+Foreman::Plugin.register :my_plugin do
+  # other code here
+  subscribe 'my_event.foreman', MyPlugin::MySubscriber
+end
+....
+
+where `my_event.foreman` is the name of the event you want to subscribe to. You may also subscribe to multiple events at once by using a regular expression, e.g. to subscribe to all events whose name ends with `.foreman` use:
+
+[source, ruby]
+....
+  subscribe /.foreman$/, MyPlugin::MySubscriber
```

Comment:
Is there any particular reason why this is suffix and not prefix? It's non-standard, I'd not expect that. I am not sure if Redis even supports that (all examples shows "TEST*" wildcard notation I need to test it).

Also I'd appreciate if this also included "event" because we might add more using Rails Subscribtion API later and this really captures the whole "namespace". So ideally: "foreman.event.NAME" if you agree with that.

Reply:
It's also a bit weird to me, but I added the namespace as suffix to follow the Rails naming convention, which is `event.library`. [[1]](https://guides.rubyonrails.org/active_support_instrumentation.html#creating-custom-events) There are two ways to subscribe to events:

- `ActiveSupport::Notifications.subscribe 'my_event.my_namespace'`, which we use currently and which I think gives more possibilities, I like that we can use regex to subscribe to multiple events at once. 
- a class that inherits from `ActiveSupport::Subscriber`, e.g.

```ruby
class MySubscriber < ActiveSupport::Subscriber
  attach_to :my_namespace

  def my_event(event)
    puts event.inspect
  end
end
```
You must attach a subscriber to the namespace and create a method named like the event you want to subscribe to. The `my_event` method will be executed whenever the `my_event.my_namespace` event is published. Note that the namespace needs to be added as suffix in this case.

Of course, as we use regex, nothing can stop us from adding namespace as a prefix. So it is possible to name events like you suggested, `foreman.event.NAME`. WDYT?

Reply:
If you followed Rails conventions than it's probably fine although not all things in Rails are great :-)